library/src-bootstrapped/scala/quoted/util/ExprMap.scala

```diff
@@ -0,0 +1,158 @@
+        }
+      }
+
+      def transformTermChildren(tree: Term, tpe: Type)(given ctx: Context): Term = tree match {
+        case Ident(name) =>
+          tree
+        case Select(qualifier, name) =>
+          Select.copy(tree)(transformTerm(qualifier, qualifier.tpe.widen), name)
+        case This(qual) =>
+          tree
+        case Super(qual, mix) =>
+          tree
+        case tree @ Apply(fun, args) =>
+          val MethodType(_, tpes, _) = fun.tpe.widen
+          Apply.copy(tree)(transformTerm(fun, defn.AnyType), transformTerms(args, tpes))
+        case TypeApply(fun, args) =>
+          TypeApply.copy(tree)(transformTerm(fun, defn.AnyType), args)
+        case _: Literal =>
+          tree
+        case New(tpt) =>
+          New.copy(tree)(transformTypeTree(tpt))
+        case Typed(expr, tpt) =>
+          val tp = tpt.tpe match
+            // TODO improve code
+            case AppliedType(TypeRef(ThisType(TypeRef(NoPrefix(), "scala")), "<repeated>"), List(IsType(tp0))) =>
+              type T
+              val a = tp0.seal.asInstanceOf[quoted.Type[T]]
+              '[Seq[$a]].unseal.tpe
+            case tp => tp
+          Typed.copy(tree)(transformTerm(expr, tp), transformTypeTree(tpt))
+        case tree: NamedArg =>
+          NamedArg.copy(tree)(tree.name, transformTerm(tree.value, tpe))
+        case Assign(lhs, rhs) =>
+          Assign.copy(tree)(lhs, transformTerm(rhs, lhs.tpe.widen))
+        case Block(stats, expr) =>
+          Block.copy(tree)(transformStats(stats), transformTerm(expr, tpe))
+        case If(cond, thenp, elsep) =>
+          If.copy(tree)(
+            transformTerm(cond, defn.BooleanType),
+            transformTerm(thenp, tpe),
+            transformTerm(elsep, tpe))
+        case _: Closure =>
+          tree
+        case Match(selector, cases) =>
+          Match.copy(tree)(transformTerm(selector, selector.tpe), transformCaseDefs(cases, tpe))
+        case Return(expr) =>
+          // FIXME
+          // ctx.owner seems to be set to the wrong symbol
+          // Return.copy(tree)(transformTerm(expr, expr.tpe))
+          tree
+        case While(cond, body) =>
+          While.copy(tree)(transformTerm(cond, defn.BooleanType), transformTerm(body, defn.AnyType))
+        case Try(block, cases, finalizer) =>
+          Try.copy(tree)(transformTerm(block, tpe), transformCaseDefs(cases, defn.AnyType), finalizer.map(x => transformTerm(x, defn.AnyType)))
+        case Repeated(elems, elemtpt) =>
+          Repeated.copy(tree)(transformTerms(elems, elemtpt.tpe), elemtpt)
+        case Inlined(call, bindings, expansion) =>
+          Inlined.copy(tree)(call, transformDefinitions(bindings), transformTerm(expansion, tpe)/*()call.symbol.localContext)*/)
+      }
+
+      def transformTerm(tree: Term, tpe: Type)(given ctx: Context): Term =
+        tree match {
+        case _: Closure =>
+          tree
+        case _: Inlined | _: Select =>
+          transformTermChildren(tree, tpe)
```

Comment:
Here we lose the ability to handle `Select` trees.

Reply:
Why?

Reply:
Removed